routers/api/v1/repo/commits.go

```diff
@@ -69,6 +69,64 @@ func GetSingleCommit(ctx *context.APIContext) {
 	ctx.JSON(http.StatusOK, json)
 }
 
+// GetSingleCommitByRef get a commit via ref
+func GetSingleCommitByRef(ctx *context.APIContext) {
```

Comment:
It seems the two function have most same codes?

Reply:
they do